geoengine/types.py

```diff
@@ -326,6 +327,17 @@ def __repr__(self) -> str:
 
         return r
 
+    # TODO: add to super class as well
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
Can't you type this better than `Any`?
The backend needs a specific type.

Reply:
I wouldn't know how, because every field has a different type. So then I'd specify a Dict with a union of all the the field value types, but we still wouldn't know which field has which type.

Reply:
But then we type everything twice right? Once in the class hierarchy and then a second time in the corresponding dicts. That doesn't seem right to me. How do we guarantee that the two hierarchies are in sync? Should we maybe use only the TypedDicts then and get rid of the classes? 

Or: instead of having to_dict methods we could have to_json methods that output something like the json::Value like in Rust. Does this exist in Python?

Reply:
Well, that is just the distinction between the API type (serialized) and the JSON type, right? This should be the same than, for instance, in the UI.

Reply:
but why do we need this distinction? The types only use is to send them to the backend. So maybe we can just define them as `TypedDicts` and just call `dict()` on them instead of implementing `to_dict` methods on dataclasses.

Reply:
I changed the dataclasses to TypedDicts: 04e3b2b514c10eeaa30770a65b6b2d4663500498